library/common/http/dispatcher.cc

```diff
@@ -234,38 +202,24 @@ Dispatcher::DirectStream::~DirectStream() {
 }
 
 void Dispatcher::DirectStream::resetStream(StreamResetReason reason) {
-  // The Http::ConnectionManager does not destroy the stream in doEndStream() when it calls
-  // resetStream on the response_encoder_'s Stream. It is up to the response_encoder_ to
-  // runResetCallbacks in order to have the Http::ConnectionManager call doDeferredStreamDestroy in
-  // ConnectionManagerImpl::ActiveStream::onResetStream.
-  //
-  // @see Dispatcher::resetStream's comment on its call to runResetCallbacks for an explanation if
-  // this if guard.
-  if (!hcm_stream_pending_destroy_) {
-    hcm_stream_pending_destroy_ = true;
-    runResetCallbacks(reason);
+  runResetCallbacks(reason);
```

Comment:
drop a comment that this is the expectation in higher layers, like the h2 codec does?